<launch>
    <!-- Run the map server -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(find project)/params/FGHFloorMap2.yaml"/>

    <!--- Run AMCL -->
    <node pkg="amcl" type="amcl" name="amcl" output="screen">
        <!--<param name="initial_pose_x" value="0"/>-->
        <!--<param name="initial_pose_y" value="0"/>-->
        <!--<param name="initial_pose_a" value="0"/>-->
        <!--26.734 9.238-->
        <param name="initial_pose_x" value="9.6"/>
        <param name="initial_pose_y" value="6.4"/>
        <!--<param name="initial_pose_a" value="1.955"/>-->
        <param name="initial_pose_a" value="-1.137"/>
        <!--<param name="initial_pose_x" value="21.0"/>-->
        <!--<param name="initial_pose_y" value="22.26"/>-->
        <!--<param name="initial_pose_a" value="2.078"/>-->
        <param name="initial_cov_xx" value="10"/>
        <param name="initial_cov_yy" value="10"/>
        <param name="initial_cov_aa" value="0.02"/>
        <param name="odom_alpha1" value="0.1"/>
        <param name="odom_alpha2" value="0.3"/>
        <param name="odom_alpha3" value="0.3"/>
        <param name="odom_alpha4" value="0.1"/>
        <param name="likelihood_max_dist" value="2.5"/>
        <!--<remap from="scan" to="scan_filtered"/>-->
    </node>
    <!--- Run rviz -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find project)/rviz/move_base_single_robot2.vcg"/>

    <param name="/use_sim_time" type="bool" value="false" />
    <!--<include file="$(find project)/launch/hokuyo_laser_filter.launch" />-->

    <!--<include file="$(find project)/launch/static_transform.launch" />-->
    <node pkg="move_base" type="move_base" respawn="false" name="p3dx_move_base" output="screen">
        <!--<remap from="scan" to="scan_filtered"/>-->
        <!--<param name="base_global_planner" value="carrot_planner/CarrotPlanner"/>-->
        <param name="recovery_behavior_enabled" value="true"/>
        <param name="conservative_reset_dist" value="4.0"/>
        <param name="clearing_rotation_allowed" value="true"/>
        <rosparam param="recovery_behaviors"> [{name: "conservative_reset", type: "clear_costmap_recovery/ClearCostmapRecovery"}, {name: "rotate_recovery", type: "rotate_recovery/RotateRecovery"}, {name: "aggressive_reset", type: "clear_costmap_recovery/ClearCostmapRecovery"}]</rosparam>
        <rosparam file="$(find project)/params/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find project)/params/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find project)/params/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find project)/params/global_costmap_params.yaml" command="load" />
        <rosparam file="$(find project)/params/base_local_planner_params.yaml" command="load" />
    </node>

    <!--Run our controller-->
    <!--<node pkg="project" type="control" name ="p3dx_control" output="screen" />-->
</launch>
